compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/tree/STMemberTypeDescriptorNode.java

```diff
@@ -0,0 +1,94 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.compiler.internal.parser.tree;
+
+import io.ballerina.compiler.syntax.tree.MemberTypeDescriptorNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NonTerminalNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * This is a generated internal syntax tree node.
+ *
+ * @since 2201.4.0
+ */
+public class STMemberTypeDescriptorNode extends STNode {
```

Comment:
Shouldn't this be STMemberNode? Similar concern for things like `SyntaxKind.MEMBER_TYPE_DESC`.

cc @rdulmina

Reply:
Another concern is `io.ballerina.compiler.syntax.tree.TupleTypeDescriptorNode#memberTypeDesc`. This doesn't quite return member typedescs now, right?

Reply:
@MaryamZi r u concerned about the name? If so `STMemberNode` is too high-level right we need to be more specific. right now `io.ballerina.compiler.syntax.tree.TupleTypeDescriptorNode#memberTypeDesc` returns `Node` since we have the rest type descriptor node to that we couldn't use `TypeDescriptorNode` as the type

Reply:
My concern is that it no longer represents just typedescs. The comparable construct for records is `STRecordFieldNode`, so IMO, would be better to have something like `STTupleMemberNode`.

I rechecked the spec though, and the spec uses member-type-descriptor, so this may be OK (https://ballerina.io/spec/lang/master/#member-type-descriptor). Just that it may be a bit confusing. WDYT?

Reply:
We usually follow the names in the spec. But I agree with u `STTupleMemberNode` is better. But it is better to align with the spec names.

Reply:
May be we can create an issue to update the spec name :smiley: WDYT?